src/test/java/de/medizininformatikinitiative/flare/model/sq/CriterionTest.java

```diff
@@ -951,61 +984,184 @@ void withTwoAttributeFilterAtFirst() {
+                            var criteria = Criterion.of(Concept.of(PARENT_TERM_CODE))
+                                    .appendAttributeFilter(AttributeFilter.ofReference(PARENT_FILTER_CODE_1,
+                                            Criterion.of(Concept.of(CHILD_TERM_CODE_1))))
+                                    .appendAttributeFilter(AttributeFilter.ofReference(PARENT_FILTER_CODE_2,
+                                            Criterion.of(Concept.of(CHILD_TERM_CODE_2)).appendAttributeFilter(
+                                                    AttributeFilter.ofConcept(CHILD_ATTR_TERM_CODE, CHILD_ATTR_VALUE))))
+                                    .expand(mappingContext);
+
+                            assertThat(criteria).isRightSatisfying(r -> assertThat(r).containsExactly(
+                                    ExpandedCriterion.of(PARENT_RESOURCE_TYPE, PARENT_CODE_SEARCH_PARAM, PARENT_TERM_CODE)
+                                            .appendFilter(new ExpandedConceptFilter(CHILD_CODE_SEARCH_PARAM_1, CHILD_TERM_CODE_1)
+                                                    .chain(PARENT_REFERENCE_FILTER_SEARCH_PARAM_1))
+                                            .appendFilter(ExpandedFilterGroup.of(
+                                                            new ExpandedConceptFilter(CHILD_CODE_SEARCH_PARAM_2, CHILD_TERM_CODE_2),
+                                                            new ExpandedConceptFilter(CHILD_ATTR_SEARCH_PARAM, CHILD_ATTR_VALUE))
+                                                    .chain(PARENT_REFERENCE_FILTER_SEARCH_PARAM_2))));
+                        }
+
+                        @Test
+                        @DisplayName("with two attribute filters at the criterion of first reference filter")
+                        void withTwoAttributeFiltersAtFirst() {
+                            when(mappingContext.findMapping(CHILD_TERM_CODE_1)).thenReturn(Either.right(Mapping.of(CHILD_TERM_CODE_1,
+                                            CHILD_RESOURCE_TYPE, CHILD_CODE_SEARCH_PARAM_1)
+                                    .appendAttributeMapping(AttributeMapping.coding(CHILD_ATTR_TERM_CODE, CHILD_ATTR_SEARCH_PARAM))
+                                    .appendAttributeMapping(AttributeMapping.coding(CHILD_ATTR_TERM_CODE_1, CHILD_ATTR_SEARCH_PARAM_1))));
+                            when(mappingContext.findMapping(CHILD_TERM_CODE_2)).thenReturn(Either.right(Mapping.of(CHILD_TERM_CODE_2,
+                                    CHILD_RESOURCE_TYPE, CHILD_CODE_SEARCH_PARAM_2)));
+
+
+                            var criteria = Criterion.of(Concept.of(PARENT_TERM_CODE))
+                                    .appendAttributeFilter(AttributeFilter.ofReference(PARENT_FILTER_CODE_1,
+                                            Criterion.of(Concept.of(CHILD_TERM_CODE_1))
+                                                    .appendAttributeFilter(AttributeFilter.ofConcept(CHILD_ATTR_TERM_CODE, CHILD_ATTR_VALUE))
```

Comment:
Please use the Constant with `_1` and `_2` suffix if you need both and without suffix if you need only one. But never the one without suffix follow by the one with the suffix `_1`.